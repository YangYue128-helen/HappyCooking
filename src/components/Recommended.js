import React, {Component} from 'react';
import {
    StyleSheet,
    View,
    TextInput,
    Text,
    FlatList,
    Dimensions,
    Image,
    ScrollView,
    TouchableOpacity
} from 'react-native';

import Ionicons from 'react-native-vector-icons/Ionicons';
import StarRating from "react-native-star-rating";

let data=[
    {
        image: require('../images/s3.png'),
        name: "Quick Turkey Taco Salad",
        time: "20mins",
        category: "Salad | Taco Salad",
        rating:4.5
    },
    {
        image: require('../images/s1.png'),
        name:"Sheet Pan Shrimp and Sausage Bake",
        time:"1hr",
        category:"Seafood | Shrimp",
        rating:5
    },
    {
        image: require('../images/s4.png'),
        name: "Oven Baked BBQ Ribs",
        time: "3hrs 30mins",
        category: "Meat and Poultry | Pork",
        rating: 4.5
    }
]

class All extends Component<{}>{
    state={
        data: data,
        data_temp: data,
        search: ''
    }
    constructor(props) {
        super(props);
    }


    _search(text) {
        let data = [];
        this.state.data_temp.map(function (recipe) {
            if(recipe.name.indexOf(text) > -1) {
                data.push(recipe);
            }
        });
        this.setState({
            data: data,
            search: text
        })
    }

    renderItem= ({item}) => {
        return(
            <TouchableOpacity onPress = { () => this.props.parentRef.props.navigation.push("RecipeDetail", {
                passon: item
            }) }>
                <View style={styles.cells}>
                    <View style={styles.recipeImage}>
                        <Image
                            style={{flex:1, width:null, height:null, resizeMode:'cover'}}
                            source = {item.image}/>
                    </View>
                    <View style={styles.recipeTextBox}>
                        <Text style={styles.recipeText}>{item.name}</Text>
                        <Text style={{fontWeight: '500', color: '#9f90b6',}}>Time: {item.time}</Text>
                        <Text style={{fontWeight: '500', color: '#9f90b6',}}>{item.category}</Text>
                        <StarRating
                            disabled={true}
                            maxStars={5}
                            rating={item.rating}
                            starSize={10}
                            fullStarColor={'#65499c'}
                        />
                    </View>
                </View>
            </TouchableOpacity>
        )
    }


    render() {
        return (
            <View style={styles.container1}>
                <View style={styles.section}>
                    <TextInput
                        placeholder="search.."
                        style={{flex: 1, marginLeft: 10}}
                        value={this.state.search}
                        onChangeText={(text) => this._search(text)}
                    />
                    <TouchableOpacity
                        onPress={()=>this._search('')}
                        style={{paddingHorizontal: 10}}
                    >
                        <Ionicons
                            name="ios-close"
                            color="gray"
                            size={20}
                        />
                    </TouchableOpacity>
                </View>
                <FlatList
                    data={this.state.data}
                    renderItem={this.renderItem}
                    keyExtractor={(item, index) => index.toString()}
                />
            </View>
        );
    }
}

export default All

var {width, height} = Dimensions.get('window')
const styles = StyleSheet.create({
    container1: {
        marginTop: 20,
        flex: 1,
        backgroundColor: '#ffffff',
    },
    section:{
        flexDirection:'row',
        alignItems: 'center',
        paddingVertical: 5,
        paddingHorizontal: 10,
        borderRadius: 100,
        backgroundColor: '#f2f2f2',
        marginTop: 0,
    },
    container: {
        marginRight: 0,
        flex: 1,
        backgroundColor: '#003300',
        alignItems: 'center',
        justifyContent: 'center',
    },
    cells:{
        marginTop: 20,
        paddingHorizontal: 20,
        width: width-60,
        height: width -100,
        borderWidth: 0.5,
        borderRadius: 25,
        backgroundColor: "#eceff1",
        borderColor: '#dddddd',
    },
    recipeImage:{
        flex: 1,
        marginTop: 10,
    },
    recipeTextBox:{
        flex: 0.7,
        alignItems:'flex-start',
        justifyContent: 'space-evenly',
        paddingLeft: 5,
    },
    recipeText:{
        fontSize: 20,
        fontWeight: 'bold',
        color: '#65499c',
    }

});