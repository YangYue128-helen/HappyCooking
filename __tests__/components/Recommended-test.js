import React from 'react';
import Recommended from "../../src/components/Recommended";

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Recommended />).toJSON();
    expect(tree).toMatchSnapshot();
});