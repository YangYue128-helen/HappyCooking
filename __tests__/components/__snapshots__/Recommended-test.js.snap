// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderRadius": 100,
        "flexDirection": "row",
        "marginTop": 0,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="search.."
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "marginLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text />
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "category": "Salad | Taco Salad",
          "image": 1,
          "name": "Quick Turkey Taco Salad",
          "rating": 4.5,
          "time": "20mins",
        },
        Object {
          "category": "Seafood | Shrimp",
          "image": 1,
          "name": "Sheet Pan Shrimp and Sausage Bake",
          "rating": 5,
          "time": "1hr",
        },
        Object {
          "category": "Meat and Poultry | Pork",
          "image": 1,
          "name": "Oven Baked BBQ Ribs",
          "rating": 4.5,
          "time": "3hrs 30mins",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#eceff1",
                "borderColor": "#dddddd",
                "borderRadius": 25,
                "borderWidth": 0.5,
                "height": 650,
                "marginTop": 20,
                "paddingHorizontal": 20,
                "width": 690,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": null,
                    "resizeMode": "cover",
                    "width": null,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 0.7,
                  "justifyContent": "space-evenly",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#65499c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Quick Turkey Taco Salad
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9f90b6",
                    "fontWeight": "500",
                  }
                }
              >
                Time: 
                20mins
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9f90b6",
                    "fontWeight": "500",
                  }
                }
              >
                Salad | Taco Salad
              </Text>
              <View
                pointerEvents="none"
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#eceff1",
                "borderColor": "#dddddd",
                "borderRadius": 25,
                "borderWidth": 0.5,
                "height": 650,
                "marginTop": 20,
                "paddingHorizontal": 20,
                "width": 690,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": null,
                    "resizeMode": "cover",
                    "width": null,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 0.7,
                  "justifyContent": "space-evenly",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#65499c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Sheet Pan Shrimp and Sausage Bake
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9f90b6",
                    "fontWeight": "500",
                  }
                }
              >
                Time: 
                1hr
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9f90b6",
                    "fontWeight": "500",
                  }
                }
              >
                Seafood | Shrimp
              </Text>
              <View
                pointerEvents="none"
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#eceff1",
                "borderColor": "#dddddd",
                "borderRadius": 25,
                "borderWidth": 0.5,
                "height": 650,
                "marginTop": 20,
                "paddingHorizontal": 20,
                "width": 690,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "flex": 1,
                    "height": null,
                    "resizeMode": "cover",
                    "width": null,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 0.7,
                  "justifyContent": "space-evenly",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#65499c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Oven Baked BBQ Ribs
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9f90b6",
                    "fontWeight": "500",
                  }
                }
              >
                Time: 
                3hrs 30mins
              </Text>
              <Text
                style={
                  Object {
                    "color": "#9f90b6",
                    "fontWeight": "500",
                  }
                }
              >
                Meat and Poultry | Pork
              </Text>
              <View
                pointerEvents="none"
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View
                  style={Object {}}
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
